src/requests.rs

```diff
@@ -0,0 +1,339 @@
+                name: username.clone(),
+                password: password.clone(),
+                role: role_value.clone(),
+            };
+            info!("Login successful");
+            Ok(login_user)
+        }
+        Err(e) => {
+            info!("Login failed");
+            Err(e.to_string())
+        }
+    }
+}
+
+/// Gets all users with the given username and password.
+///
+/// # Arguments
+///
+/// * `username` - A string slice that holds the username.
+/// * `password` - A string slice that holds the password.
+///
+/// # Returns
+///
+/// Returns a vector of `User` structs representing all the users.
+pub async fn get_all_users(username: String, password: String) -> RequestResult<Vec<User>> {
+    let client = reqwest::Client::new();
+    let response = client
+        .get(ENDPOINT.to_string() + "users")
+        .header(
+            "Authorization",
+            "Basic ".to_string() + &encode_credentials(username.clone(), password.clone()),
+        )
+        .send()
+        .await
+        .map_err(|e| e.to_string())?;
+
+    let result = response.error_for_status_ref().map(|_| ());
+
+    match result {
+        Ok(_) => {
+            let ids: Vec<i64> = response.json().await.map_err(|e| e.to_string())?;
+
+            let mut users = Vec::new();
+            for id in ids {
+                let response = client
+                    .get(ENDPOINT.to_string() + &format!("user/{}", id))
+                    .header(
+                        "Authorization",
+                        "Basic ".to_string()
+                            + &encode_credentials(username.clone(), password.clone()),
+                    )
+                    .send()
+                    .await
+                    .map_err(|e| e.to_string())?;
+
+                let temp_user: TempUser = response.json().await.map_err(|e| e.to_string())?;
+
+                let role = PlantBuddyRole::try_from(temp_user.role).unwrap();
+                let user = User {
+                    id: temp_user.id,
+                    name: temp_user.name,
+                    role,
+                    password: String::new(),
+                };
+
+                users.push(user);
+            }
+            info!("Get all users successful");
+            Ok(users)
+        }
+        Err(e) => {
+            info!("Get all users failed");
+            Err(e.to_string())
+        }
+    }
+}
+
+/// Creates a new user with the given username, password, and user data.
+///
+/// # Arguments
+///
+/// * `username` - A string slice that holds the username.
+/// * `password` - A string slice that holds the password.
+/// * `user` - A `TempCreationUser` struct representing the user to create.
+///
+/// # Returns
+///
+/// Returns a `RequestResult` indicating whether the user was created successfully.
+pub async fn create_user(
+    username: String,
+    password: String,
+    user: TempCreationUser,
+) -> RequestResult<()> {
+    let client = reqwest::Client::new();
```

Comment:
And the client could share the reqwest::Client too, which would make the API calls much more efficient since name resolution would not happen on a per-request basis